### PR TITLE
Updated login screen to compose and added some comments from previous PR

### PR DESCRIPTION
**Pull request by ianslinker-br**
So despite the fact that bitbucket has shut down this method of login as of today, here is an updated version of the login screen in compose.

The majority of this was pretty straightforward, but I made some changes to how the screen is laid out in order to reduce the use of large margins to "center" things. The text fields were the most complex part of the screen as compose does not do any of the focus work for you. Additionally, the default behavior for text fields does not change the text or icon color when focused so I had to keep track of the focus in a state val and change those myself. And as a final twist, the show hide password behavior had to be coded manually as well. 

The lock icon on zeplin was not available as an svg, so I switched to using the one provided by Android.

### CODE REVIEW

**Review by BottleRocket-Colin — APPROVED** [review]
WOOT!

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt [hunk 1]
```diff
@@ -1,16 +1,316 @@
...
+    val onSignupClicked: () -> Unit,
+    val onForgotClicked: () -> Unit,
+    val onDevOptionsClicked: () -> Unit,
+    val onEmailChanged: (String) -> Unit,
+    val onPasswordChanged: (String) -> Unit
 )
 
 @Composable
 fun LoginScreen(state: LoginScreenState) {
+    Surface(
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt:61]: @ianslinker-br Should the root screen view be some scrolling container to allow for the screen to be scrollable when the soft keyboard is up?

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt [hunk 2]
```diff
@@ -1,16 +1,316 @@
...
+                .padding(
+                    start = Dimens.grid_6,
+                    end = Dimens.grid_6,
+                    top = Dimens.grid_2
+                )
+                .fillMaxWidth(),
+            valueText = state.email.value,
+            onValueChange = state.onEmailChanged,
+            onNextClicked = {
+                localFocusManager.moveFocus(FocusDirection.Down)
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt:145]: @ianslinker-br This is an interesting difference from how xml layouts handle focus. Thanks for maintaining user expectations!

File: compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt [hunk 3]
```diff
@@ -3,6 +3,17 @@ package com.bottlerocketstudios.compose.resources
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
 
+// https://proandroiddev.com/supporting-different-screen-sizes-on-android-with-jetpack-compose-f215c13081bd
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/resources/Dimens.kt:6]: @ianslinker-br Great writeup and thanks for the link!

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt (hunk 1 above)
Comment (ianslinker-br) [reply compose/src/main/java/com/bottlerocketstudios/compose/auth/Login.kt:61]: In my testing, this still allows the screen to be scrollable when the keyboard is up. The wrapped column content becomes larger than the surface size and the vertical scroll applied to it allows the user to scroll the content as necessary.

**Review by jaredstockton — APPROVED** [review]
Nice job and appreciate the PR description as always!

**Pull request merged**